### PR TITLE
AArch64: Implement OMRMemoryReference::assignRegisters()

### PR DESCRIPTION
**Pull request by knn-k**
This commit implements assignRegisters() in OMRMemoryReference
for aarch64.

Signed-off-by: knn-k <konno@jp.ibm.com>

### CODE REVIEW

**Comment by knn-k** [comment]
Updated OMRMemoryReference.hpp by adding some functions to it.

File: compiler/aarch64/codegen/OMRMemoryReference.hpp [hunk 1]
```diff
@@ -60,6 +60,8 @@ class OMR_EXTENSIBLE MemoryReference : public OMR::MemoryReference
    TR::Node *_indexNode;
    int32_t _offset;
 
+   TR::Register *_modBase;
```
Comment (0xdaryl) [inline compiler/aarch64/codegen/OMRMemoryReference.hpp]: Can you explain what `_modBase` is used for?  One-third of your assignment logic is dedicated to assigning it, but its not immediately clear to me what its for.

File: compiler/aarch64/codegen/OMRInstruction.cpp [hunk 2]
```diff
@@ -52,6 +52,13 @@ OMR::ARM64::Instruction::remove()
    }
 
 
+void OMR::ARM64::Instruction::ARM64NeedsGCMap(uint32_t mask)
+   {
+   if (TR::comp()->useRegisterMaps())
```
[the remark this answers is not in the document]
Comment (knn-k) [reply compiler/aarch64/codegen/OMRInstruction.cpp]: Do you mean passing `Compilation` pointer here?  `useRegisterMaps()` is defined only in `Compilation`.

File: compiler/aarch64/codegen/OMRMemoryReference.hpp (hunk 1 above)
Comment (0xdaryl) [reply compiler/aarch64/codegen/OMRMemoryReference.hpp]: Yes, unless you can explain what it will be used for in AArch64.

**Comment by 0xdaryl** [comment]
This is an AArch64-specific change.  No need to wait for the AppVeyor build.  Merging.

**Pull request merged**